### PR TITLE
Added --skip-pull, --config, --substitution flags, closes #913. 

### LINKED ISSUE #913: cli: add more options to popper sh
We can add the `--skip-pull`, `--substitution`, `--config`, etc. flags to `popper sh` to enhance usability.

### PR DESCRIPTION
**Pull request by maruthgoyal**
Closes #913 , and adds style fixes using black, as specified in the contributing guidelines.

### CODE REVIEW

**Comment by maruthgoyal** [comment]
I used `black .` for formatting, and `black --check --diff .` has the following output

> All done! ✨ 🍰 ✨
> 34 files would be left unchanged.

However, it's erroring on Travis. Any suggestions?

**Comment by JayjeetAtGithub** [comment]
Hi @maruthgoyal. Thanks a lot for working on this issue. The changes look great.

Regarding the Travis errors, I presume you are working with the latest version of black i.e. `20.8b1`. Did you run `pip install -e src/[dev]`?. This should install the correct version of black.

We use the `19.10b0` version (see [here](https://github.com/getpopper/popper/blob/master/src/setup.py#L32)) and that's why the formatting in your PR is different. From black's release history, it looks like they are close to making a stable release soon. I think we can continue to do the formatting using the `19.10b0` version till then to avoid changing the format of the entire codebase with every beta release.

**Comment by maruthgoyal** [comment]
Hi @JayjeetAtGithub, thank you for that. My apologies on missing that detail. I believe it passes the checks now

File: src/popper/commands/cmd_version.py [hunk 1]
```diff
@@ -7,8 +7,7 @@
 @click.command("version", short_help="Show version of Popper and exit.")
 @pass_context
 def cli(ctx):
-    """Displays version of Popper and exit.
-    """
+    """Displays version of Popper and exit."""
```
Comment (JayjeetAtGithub) [inline src/popper/commands/cmd_version.py:10]: @maruthgoyal Are these changes done by black? Same for the other files. I guess not. You can just make the changes to the `cmd_sh.py` file and make the PR.

Comment (maruthgoyal) [reply src/popper/commands/cmd_version.py]: Hey Jayjeet, apologies I believe those changes were made by the run of the previous version of black. I have restored the file's respective states to that of the latest commit on master.

Comment (JayjeetAtGithub) [reply src/popper/commands/cmd_version.py]: No worries. Thanks !

**Comment by JayjeetAtGithub** [comment]
LGTM !

**Comment by ivotron** [comment]
thanks a lot @maruthgoyal , this looks great. Would it be possible to also add `--skip-clone`?

**Review by ivotron — APPROVED** [review]
lgtm!

**Comment by ivotron** [comment]
thanks @maruthgoyal! 🙏

**Pull request merged**